services/network/network.go

```diff
@@ -24,6 +24,14 @@ type SubResource struct {
 	ID *string `json:"id,omitempty"`
 }
 
+// GroupSubResource referes to another subresource, and the group that subresource belongs to
+type GroupSubResource struct {
```

Comment:
can you just briefly explain here why are we referencing the resource differently here?

Reply:
Sorry didn't see this earlier! We did have this previously because we wanted to reference an NSG in an LNET, but we coulnd't just use the NSG name because NSG was parented to group while LNET was parented to location, so we couldn't use the LNET's group. That said, because of this problem we decided to change to have NSG parented at the location level - so this was removed again